crates/bevy_gizmos/src/primitives/dim2.rs

```diff
@@ -467,9 +467,9 @@ impl<'w, 's, const N: usize, T: GizmoConfigGroup> GizmoPrimitive2d<Polygon<N>>
 
         // Check if the polygon needs a closing point
         let closing_point = {
-            let last = primitive.vertices.last();
-            (primitive.vertices.first() != last)
-                .then_some(last)
+            let first = primitive.vertices.first();
```

Comment:
This looks like a bug fix? Can you explain more and/or split this out?

Reply:
Yeah, indeed, I'll split it into another PR :+1:

I tried explaining it in the commit a bit https://github.com/bevyengine/bevy/pull/11697/commits/a9931ad96421a7db25e669bfa818dd86214ca577

Reply:
Ah, should have checked the commits 😅 Looks like a solid fix, but yeah, let's split this out and merge it.

Reply:
Done in #11699